arm/Microsoft.Storage/storageAccounts/.bicep/nested_fileShare_rbac.bicep

```diff
@@ -1,12 +1,12 @@
 param fileShareName string
-param roleAssignment object
+param roleAssignmentObj object
 param builtInRoleNames object
 param storageAccountName string
 
-resource nested_fileShare_rbac 'Microsoft.Storage/storageAccounts/fileServices/fileshares/providers/roleAssignments@2018-09-01-preview' = [for principalId in roleAssignment.principalIds: {
-  name: '${storageAccountName}/default/${fileShareName}/Microsoft.Authorization/${(empty(roleAssignment) ? guid(storageAccountName) : guid(storageAccountName, fileShareName, principalId, roleAssignment.roleDefinitionIdOrName))}'
+resource nested_fileShare_rbac 'Microsoft.Storage/storageAccounts/fileServices/fileshares/providers/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignmentObj.principalIds: {
```

Comment:
resource roleAssignment

Reply:
Fixed, resolve if you agree.

Reply:
I'd replace `nested_fileShare_rbac` with `roleAssignment` but let's take all storage account alignment in a different PR